### ISSUE TITLE
Upgrade dependency`nan` of pty.js ?

### ISSUE THREAD

**Issue by luxrck** [platformio/platformio-atom-ide-terminal#79]
```
gyp info it worked if it ends with ok
gyp info using node-gyp@3.4.0
gyp info using node@6.7.0 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/luxrck/.atom/packages/platformio-ide-terminal/node_modules/pty.js/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/luxrck/.node-gyp/6.7.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/luxrck/.node-gyp/6.7.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/luxrck/.atom/packages/platformio-ide-terminal/node_modules/pty.js',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/luxrck/.atom/packages/platformio-ide-terminal/node_modules/pty.js/build'
  CXX(target) Release/obj.target/pty/src/unix/pty.o
In file included from ../node_modules/nan/nan.h:184:0,
                 from ../src/unix/pty.cc:20:
../node_modules/nan/nan_maybe_43_inl.h: In function ‘Nan::MaybeLocal<v8::Object> Nan::CloneElementAt(v8::Local<v8::Array>, uint32_t)’:
../node_modules/nan/nan_maybe_43_inl.h:221:58: warning: ‘v8::MaybeLocal<v8::Object> v8::Array::CloneElementAt(v8::Local<v8::Context>, uint32_t)’ is deprecated: Cloning is not supported. [-Wdeprecated-declarations]
   return array->CloneElementAt(GetCurrentContext(), index);
                                                          ^
In file included from /home/luxrck/.node-gyp/6.7.0/include/node/v8.h:25:0,
                 from /home/luxrck/.node-gyp/6.7.0/include/node/node.h:42,
                 from ../node_modules/nan/nan.h:24,
                 from ../src/unix/pty.cc:20:
/home/luxrck/.node-gyp/6.7.0/include/node/v8.h:3032:36: note: declared here
                 MaybeLocal<Object> CloneElementAt(Local<Context> context,
                                    ^
/home/luxrck/.node-gyp/6.7.0/include/node/v8config.h:333:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../node_modules/nan/nan_new.h:189:0,
                 from ../node_modules/nan/nan.h:190,
                 from ../src/unix/pty.cc:20:
../node_modules/nan/nan_implementation_12_inl.h: In static member function ‘static Nan::imp::FactoryBase<v8::BooleanObject>::return_t Nan::imp::Factory<v8::BooleanObject>::New(bool)’:
../node_modules/nan/nan_implementation_12_inl.h:40:38: warning: ‘static v8::Local<v8::Value> v8::BooleanObject::New(bool)’ is deprecated: Pass an isolate [-Wdeprecated-declarations]
   return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                                      ^
In file included from /home/luxrck/.node-gyp/6.7.0/include/node/v8.h:25:0,
                 from /home/luxrck/.node-gyp/6.7.0/include/node/node.h:42,
                 from ../node_modules/nan/nan.h:24,
                 from ../src/unix/pty.cc:20:
/home/luxrck/.node-gyp/6.7.0/include/node/v8.h:4007:56: note: declared here
   V8_DEPRECATED("Pass an isolate", static Local<Value> New(bool value));
                                                        ^
/home/luxrck/.node-gyp/6.7.0/include/node/v8config.h:333:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/unix/pty.cc:20:0:
../node_modules/nan/nan.h: At global scope:
../node_modules/nan/nan.h:592:20: error: variable or field ‘AddGCEpilogueCallback’ declared void
       v8::Isolate::GCEpilogueCallback callback
                    ^~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:592:7: error: ‘GCEpilogueCallback’ is not a member of ‘v8::Isolate’
       v8::Isolate::GCEpilogueCallback callback
       ^~
../node_modules/nan/nan.h:593:18: error: expected primary-expression before ‘gc_type_filter’
     , v8::GCType gc_type_filter = v8::kGCTypeAll) {
                  ^~~~~~~~~~~~~~
../node_modules/nan/nan.h:598:20: error: variable or field ‘RemoveGCEpilogueCallback’ declared void
       v8::Isolate::GCEpilogueCallback callback) {
                    ^~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:598:7: error: ‘GCEpilogueCallback’ is not a member of ‘v8::Isolate’
       v8::Isolate::GCEpilogueCallback callback) {
       ^~
../node_modules/nan/nan.h:603:20: error: variable or field ‘AddGCPrologueCallback’ declared void
       v8::Isolate::GCPrologueCallback callback
                    ^~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:603:7: error: ‘GCPrologueCallback’ is not a member of ‘v8::Isolate’
       v8::Isolate::GCPrologueCallback callback
       ^~
../node_modules/nan/nan.h:604:18: error: expected primary-expression before ‘gc_type_filter’
     , v8::GCType gc_type_filter = v8::kGCTypeAll) {
                  ^~~~~~~~~~~~~~
../node_modules/nan/nan.h:609:20: error: variable or field ‘RemoveGCPrologueCallback’ declared void
       v8::Isolate::GCPrologueCallback callback) {
                    ^~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:609:7: error: ‘GCPrologueCallback’ is not a member of ‘v8::Isolate’
       v8::Isolate::GCPrologueCallback callback) {
       ^~
../node_modules/nan/nan.h: In function ‘bool Nan::SetAccessor(v8::Local<v8::Object>, v8::Local<v8::String>, Nan::GetterCallback, Nan::SetterCallback, v8::Local<v8::Value>, v8::AccessControl, v8::PropertyAttribute)’:
../node_modules/nan/nan.h:1939:16: warning: ‘bool v8::Object::SetAccessor(v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::Local<v8::Value>, v8::AccessControl, v8::PropertyAttribute)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     , attribute);
                ^
In file included from /home/luxrck/.node-gyp/6.7.0/include/node/v8.h:25:0,
                 from /home/luxrck/.node-gyp/6.7.0/include/node/node.h:42,
                 from ../node_modules/nan/nan.h:24,
                 from ../src/unix/pty.cc:20:
/home/luxrck/.node-gyp/6.7.0/include/node/v8.h:2751:22: note: declared here
                 bool SetAccessor(Local<Name> name,
                      ^
/home/luxrck/.node-gyp/6.7.0/include/node/v8config.h:333:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
make: *** [pty.target.mk:94: Release/obj.target/pty/src/unix/pty.o] Error 1
make: Leaving directory '/home/luxrck/.atom/packages/platformio-ide-terminal/node_modules/pty.js/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Linux 4.7.5-1-ARCH
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/luxrck/.atom/packages/platformio-ide-terminal/node_modules/pty.js
gyp ERR! node -v v6.7.0
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
```

This caused #68 on archlinux with `node v6.7.0`
The prebuilt version package`atom-editor-bin`contains it's own copy of node, which is`node v4.4.5`
Update`nan` to the latest version solves #68 on archlinux

**Comment by sprat** [comment]
Same problem here. The nan compilation problem has been solved upstream, so it must be upgraded, and problably pty.js too since it has a dependency to nan.

**Comment by luxrck** [comment]
This package uses platformio's own fork of pty.js, which currently use `nan v2.0.5`.
https://github.com/platformio/pty.js/blob/prebuilt/package.json

**Comment by fusion809** [comment]
Seems like a duplicate of [tensor5/arch-atom#12](https://github.com/tensor5/arch-atom/issues/12). The solution is to use my package, [`terminal-fusion`](https://github.com/fusion809/terminal-fusion). It uses my own fork of pty.js, with `nan@2.3.5`.

**Comment by sprat** [comment]
The idea of the nan/pty update was already discussed here: https://github.com/platformio/platformio-atom-ide-terminal/pull/39. The issue is closed but the problem is still there.

**Comment by sprat** [comment]
It's probably due to an incompatiblity with node v6, shipped with Arch Linux by default, and causing problems with many other plugins too.

**Comment by ivankravets** [comment]
Fixed in the latest 2.2.1 version that has been just released. Should work on each machine now.

@luxrck, @sprat, @fusion809  could you confirm that? Please update package to 2.2.1. This version doesn't have PTY binary for Atom 1.12. If you have `gcc` inіtalled, Atom will build it automatically for you.

**Comment by fusion809** [comment]
@ivankravets Seems to work for me.

**Comment by ivankravets** [comment]
Resolved in 2.2.1
